libcudacxx/test/libcudacxx/cuda/barrier/cp_async_bulk_ptx_compiles.pass.cpp

```diff
@@ -52,14 +58,20 @@ __global__ void test_bulk_tensor(CUtensorMap* map)
 __global__ void test_bulk(void* gmem)
 {
   __shared__ int smem;
-  __shared__ barrier* bar;
-  if (threadIdx.x == 0)
+#if _CCCL_CUDA_COMPILER(CLANG)
+  // clang-cuda complains about initialization of a variable in shared memory
+  alignas(8) __shared__ char barrier_data[sizeof(barrier)];
+  barrier& bar = *reinterpret_cast<barrier*>(&barrier_data);
+#else // ^^^ _CCCL_CUDA_COMPILER(CLANG) ^^^ / vvv !_CCCL_CUDA_COMPILER(CLANG)
+  __shared__ barrier bar;
+#endif // !_CCCL_CUDA_COMPILER(CLANG)
+  //   if (threadIdx.x == 0)
   {
-    init(bar, blockDim.x);
+    init(&bar, blockDim.x);
```

Comment:
This looks like a bug. In my understanding, a barrier should only be initialized once from a single thread. Please reinstate the `if (threadIdx.x == 0)`. Why was it commented out?

Reply:
I should not open PRs on Fridays...